app/uk/gov/hmrc/pillar2submissionapi/models/uktrsubmissions/UktrSubmission.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.pillar2submissionapi.models.uktrsubmissions
+
+import play.api.libs.json.{JsValue, Json, OFormat, Reads}
+
+trait UktrSubmission {
+
+  implicit val uktrSubmissionDataFormat:      OFormat[UktrSubmissionData]      = Json.format[UktrSubmissionData]
+  implicit val uktrSubmissionNilReturnFormat: OFormat[UktrSubmissionNilReturn] = Json.format[UktrSubmissionNilReturn]
+
+//  implicit val userRequestReads: Reads[UktrSubmission] = (json: JsValue) =>
+//    (json \ "liabilities").asOpt[ReturnType] match {
+//      case Some(nilReturnRequest) =>
+//        if (nilReturnRequest.entryName == ReturnType.NilReturn.entryName) {
+//          UktrSubmissionNilReturn.format.reads(json)
+//        } else ???
+//      case None =>
+//        UktrSubmissionData.format.reads(json)
+//    }
+}
```

Comment:
I think it's be to have the read/write logic here in a companion object, also it tends to be good practice to have our common values in the trait itself

```suggestion
sealed trait UktrSubmission {
  val accountingPeriodFrom: String
  val accountingPeriodTo: String
  val qualifyingGroup: Boolean
  val obligationDTT: Boolean
  val obligationMTT: Boolean
}

object UktrSubmission {
  implicit val reads: Reads[UktrSubmission] = (json: JsValue) =>
    (json \ "liabilities").asOpt[LiabilityNilReturn] match {
      case Some(nilReturnRequest) =>
        if (nilReturnRequest.returnType == ReturnType.NilReturn.entryName) {
          UktrSubmissionNilReturn.format.reads(json)
        } else JsError("Invalid return type.")
      case None =>
        UktrSubmissionData.format.reads(json)
    }

  implicit val writes: Writes[UktrSubmission] = {
    case nil: UktrSubmissionNilReturn => Json.writes[UktrSubmissionNilReturn].writes(nil)
    case data: UktrSubmissionData => Json.writes[UktrSubmissionData].writes(data)
  }

  implicit val format: Format[UktrSubmission] = Format(reads, writes)
}
```